sqlglot/dialects/duckdb.py

```diff
@@ -852,3 +853,18 @@ def objectinsert_sql(self, expression: exp.ObjectInsert) -> str:
                 return self.func("STRUCT_PACK", kv_sql)
 
             return self.func("STRUCT_INSERT", this, kv_sql)
+
+        def generatedatearray_sql(self, expression: exp.GenerateDateArray) -> str:
+            start = _implicit_date_cast(expression.args.get("start"))
+            end = _implicit_date_cast(expression.args.get("end"))
+
+            # BigQuery has a default step/interval of 1 DAY
+            interval = expression.args.get("interval") or exp.Interval(
+                this=exp.Literal.number(1), unit=exp.var("DAY")
+            )
```

Comment:
Is this only a BQ concept? Should we be doing this AST normalization @ parse time for BQ?

Reply:
Yeah afaict this function is only in BQ, although it was introduced here https://github.com/tobymao/sqlglot/pull/3169/files without further context or BQ tests.

The parsing is now performed implicitly but we can definitely add an entry to `FUNCTIONS` and normalize it there as well

Reply:
Sure we can do this normalization by adding a `Parser.FUNCTIONS` entry.